### DISCUSSION TITLE
Read files from filesystem when deployed to vercel

### DISCUSSION THREAD

**Discussion by muellercornelius** [sveltejs/kit#7394] [Q&A] [answered]
Hi Guys ;),

I have a small api endpoint which reads files from the public folder and just sends them back. The problem is when deployed to vercel the folder structure changes and I always get invalid path... I have deployed already ten different versions with different solutions to find out the path. 
import {promises as fs} from "fs";
import path from "path";
```js
export async function GET() {
    const imagePath = path.join(import.meta.url, 'static', 'image.jpg')
    const asset = await fs.readFile(imagePath);
    return new Response(asset, {
        headers: {
            "Content-Type": "image/jpg"
        }
    })
}
```
I tried `process.cwd()`... but maybe that would have been correct and i just missed a small part. My error messages in the vercel console were always something like that: 

`Error: ENOENT: no such file or directory, open '/var/task/image.jpg'` 
the code for this error message was: 
```js
const imagePath = path.resolve(process.cwd(), 'image.jpg')
```

Thank you for your help ;).

Have a nice day 😉

**Comment by bartvdbraak** [comment]
Did you ever fix this issue?

**Comment by teemingc** [comment, marked as answer]
Update for all: use the new `read` [API](https://kit.svelte.dev/docs/modules#$app-server-read) that simplifies reading files from the filesystem / deployment environment.

Watch an explanation from Rich here https://www.youtube.com/watch?v=m4G-6dyF1MU
